cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/LowResourceUtilizationTest.java

```diff
@@ -0,0 +1,186 @@
+import com.linkedin.kafka.cruisecontrol.config.constants.MonitorConfig;
+import com.linkedin.kafka.cruisecontrol.exception.OptimizationFailureException;
+import com.linkedin.kafka.cruisecontrol.executor.ExecutionProposal;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+import com.linkedin.kafka.cruisecontrol.model.ReplicaPlacementInfo;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.kafka.common.TopicPartition;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Unit test for testing low utilization threshold taking effect to prevent rebalance for resource distribution goals
+ */
+@RunWith(Parameterized.class)
+public class LowResourceUtilizationTest {
+
+  private static final Map<Resource, Double> BROKER_CAPACITY = new HashMap<Resource, Double>() {{
+    put(Resource.CPU, 200.0);
+    put(Resource.DISK, 1000.0);
+    put(Resource.NW_IN, 2000.0);
+    put(Resource.NW_OUT, 2000.0);
+  }};
+
+  private final ClusterModel _clusterModel;
+  private final ResourceDistributionGoal _resourceDistributionGoal;
+  private final Map<String, Object> _goalConfigs;
+  private final Boolean _expectRebalance;
+
+  public LowResourceUtilizationTest(ClusterModel clusterModel,
+                                    ResourceDistributionGoal resourceDistributionGoal,
+                                    Map<String, Object> goalConfigs,
+                                    Boolean expectRebalance) {
+
+    _clusterModel = clusterModel;
+    _resourceDistributionGoal = resourceDistributionGoal;
+    _goalConfigs = goalConfigs;
+    _expectRebalance = expectRebalance;
+  }
+
+  /**
+   * Populate parameters to test rebalance with low utilization threshold. All brokers are alive.
+   *
+   * @return Parameters to test rebalance with low utilization threshold.
+   */
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    Collection<Object[]> p = new ArrayList<>();
+
+    // Expect rebalance on CPU usage distribution goal since one broker's CPU usage is above the low CPU utilization threshold
+    ClusterModel clusterModel = createSmallClusterModel();
+    Map<String, Object> goalConfigs = getDefaultGoalConfigs();
+    goalConfigs.put(AnalyzerConfig.CPU_LOW_UTILIZATION_THRESHOLD_CONFIG, 0.3474);
+    p.add(params(clusterModel, new CpuUsageDistributionGoal(), goalConfigs, true));
+
+    // Expect no rebalance on CPU usage distribution goal since all brokers' CPU usage is below the low CPU utilization threshold
+    clusterModel = createSmallClusterModel();
+    goalConfigs = getDefaultGoalConfigs();
+    goalConfigs.put(AnalyzerConfig.CPU_LOW_UTILIZATION_THRESHOLD_CONFIG, 0.3862);
```

Comment:
Can we explain in comments how we came up with these numbers? Looks like `0.3862` is much higher than the highest broker utilization (i.e. `34.75`). Why do we add this additional buffer?

Reply:
I changed the implementation so that the calculation process of these numbers are clear